### ISSUE TITLE
Log learning rate to tensorboard summaries

### ISSUE THREAD

**Issue by udaypk** [google-research/bert#484]
I was trying to add learning rate to tensorboard summaries but could not figure out how to do that. Even for loss_1 and loss_2 that are already logged to tensorboard events I could not find where they are done (I could not see tf.summary anywhere in the code). Can someone please help with this.

**Comment by hsm207** [comment]
You just need to add a call to tf.summary.scalar().

Check [this ](https://github.com/hsm207/bert_attn_viz/blob/6f8b1a2a930230bf25ef709721c00d28b0a89688/bert_attn_viz/optimization.py#L25-L29)out for an example:

**Comment by udaypk** [comment]
Thanks for the quick response. But I'm getting the following error (i'm using the :

INFO:tensorflow:Cannot use 'learning_rate' as input to 'Merge_2/MergeSummary' because 'learning_rate' is in a while loop.
Merge_2/MergeSummary while context: None
learning_rate while context: while_context

**Comment by udaypk** [comment]
TPUEstimator documentation (https://cloud.google.com/tpu/docs/tutorials/migrating-to-tpuestimator-api in section Update the custom model function) says tf.summary is not supported. Were you able to workaround this restriction.

**Comment by hsm207** [comment]
My code is based on the the same repo too but an older version. I am using Tensorflow 1.12 and do not get the error you reported.

**Comment by kenkrige** [comment]
> You just need to add a call to tf.summary.scalar().
> 
> Check [this ](https://github.com/hsm207/bert_attn_viz/blob/6f8b1a2a930230bf25ef709721c00d28b0a89688/bert_attn_viz/optimization.py#L25-L29)out for an example:
I am really strugling to get this right. I have modified run_classifier.py for a current kaggle competition, and it is getting great results, but I'm finding it difficult to monitor. I have littered the code with tf.summary (.scalar and .histogram) statements but when I run tensorboard on the output directory, only the projection works.

**Comment by hsm207** [comment]
@kenkrige why not try cloning my repo, implement your modifications and see if it works?

**Comment by kenkrige** [comment]
Thanks @hsm207 I might try that. Does the modified run_classifier in your repo publish results to tensorboard while it is training, even on a GPU? I am running into the problem mentioned by @udaypk of TPUEstimator not publishing properly when it runs on a GPU.

**Comment by hsm207** [comment]
@kenkrige yes, it publishes results to tensorboard even on a GPU.

**Comment by deepset-ai** [comment]
Has anybody found a workaround to store more detailed summaries for tensorboard even during TPU training?
